darts/timeseries.py

```diff
@@ -911,6 +935,160 @@ def univariate_values(self, copy=True, sample=0) -> np.ndarray:
+        axis = TimeSeries._get_str_axis(axis)
+        try:
+            return TimeSeries(self._xa[{axis: range(-samples, 0)}])
+        except IndexError:
+            return self
+
+    @staticmethod
+    def _get_str_axis(axis: Union[int, str]):
+        """Returns correct axis name as a string, regardless whether it was string or integer"""
+        axis_error_string = "Axis parameter can be only one of the numbers (0, 1, 2) " \
+                            "or strings ('time', 'component', 'sample')"
+        if isinstance(axis, int):
+            try:
+                axis = DIMS[axis]
+            except IndexError:
+                raise AttributeError(axis_error_string)
+        elif isinstance(axis, str):
+            if axis not in DIMS:
+                raise AttributeError(axis_error_string)
+        else:
+            raise AttributeError(axis_error_string)
+
+        return axis
+
+    @staticmethod
+    def concatenate(timeserie_sequence: Sequence['TimeSeries'],
+                    axis: Optional[Union[str, int]] = 0,
+                    ignore_time_axes: Optional[bool] = False):
+        """Concatenates other timeseries to the current one along given axis.
+
+            Note: when concatenating along the ``time`` dimension, first concatenated timeserie marks the start date of
+            concatenated series and remaining series will have their date indices overwritten.
+
+            Parameters
+            ----------
+            timeserie_sequence : sequence of TimeSeries
+                another timeseries to concatenate to this one
+            axis : str or int
+                axis along which timeseries will be concatenated. ['time', 'component' or 'sample'; Default: 'time']
+            ignore_time_axes : bool, default False
+                Ignore errors when time axis is varies for some time series. Note that this may yield unexpected results
+
+            Returns
+            -------
+            TimeSeries
+                concatenated timeseries
+        """
+
+        axis = TimeSeries._get_str_axis(axis)
+        if len(timeserie_sequence) < 2:
+            return timeserie_sequence
+
+        start_date_equal = len(set([ts.start_time() for ts in timeserie_sequence])) == 1
+        end_date_equal = len(set([ts.end_time() for ts in timeserie_sequence])) == 1
+        freq_equal = len(set([ts.freq_str for ts in timeserie_sequence])) == 1
+        time_axis_equal = start_date_equal and end_date_equal and freq_equal
+        component_axis_equal = len(set([ts.width for ts in timeserie_sequence])) == 1
+        sample_axis_equal = len(set([ts.n_samples for ts in timeserie_sequence])) == 1
+
+        if ((axis == 'time' and not (component_axis_equal and sample_axis_equal)) or
+            (axis == 'component' and not ((time_axis_equal or ignore_time_axes) and sample_axis_equal)) or
+            (axis == 'sample' and not ((time_axis_equal or ignore_time_axes) and component_axis_equal))):
```

Comment:
Yes it's important, there are cases where it could fail otherwise.

Reply:
Actually for the name of the time axis you should use `TimeSeries._time_dim` because we allow this one to have a different name.

Reply:
I can't use `TimeSeries._time_dim` because it's `@staticmethod`, and since you have suggested moving it out from `TimeSeries` class, I won't be able to use it either. Unless I am missing something, @hrzn? For now I used suggestion from @dennisbader.